### PR TITLE
fix: Fix changelog generation in tag-mode (#171)

### LINKED ISSUE #171: Changelog generation broken in tag mode
*Splitting https://github.com/relekang/python-semantic-release/issues/109#issuecomment-586504170 off into a separate issue.*

I'm having some trouble getting release notes to publish to GitHub. Here's the relevant section from the debug log:

```
Updating changelog
New release published
Wed, 12 Feb 2020 23:34:00 GMT semantic_release:hvcs 'get_hvcs: hvcs=' 'github' +7.1s
Wed, 12 Feb 2020 23:34:00 GMT semantic_release:history:logs 'generate_changelog("0.4.0", "1.0.0")' +7.1s
Wed, 12 Feb 2020 23:34:00 GMT semantic_release:history:logs 'markdown_changelog(version="1.0.0", header=False, changelog=...)' +4.2ms
Wed, 12 Feb 2020 23:34:00 GMT semantic_release:hvcs 'post_changelog(owner=AlphaMycelium, repository=numunit, version=1.0.0)' +4.4ms
Wed, 12 Feb 2020 23:34:00 GMT semantic_release:hvcs 'get_hvcs: hvcs=' 'github' +71.75us
Wed, 12 Feb 2020 23:34:00 GMT semantic_release:hvcs:github 'listing releases' +0.0us
Wed, 12 Feb 2020 23:34:01 GMT semantic_release:hvcs:github 'response #1, status_code=201, status=True' +235.7ms
```

The created release does not contain the Markdown changelog but instead just the version number.

### PR DESCRIPTION
**Pull request by betaboon**
resolves #171 

i tested this locally.

the whole-testsuite seems to still pass.

i have no clue how to test this with the given test-infrastructure tho.

suggestions and contributions to cover this with tests are highly welcome

### CODE REVIEW

File: semantic_release/vcs_helpers.py [hunk 1]
```diff
@@ -37,18 +37,34 @@ def get_formatted_tag(version):
...
     if from_rev:
         from_rev = get_formatted_tag(from_rev)
         try:
             repo().commit(from_rev)
-            rev = f"...{from_rev}"
         except BadName:
+            from_rev = None
             logger.debug(
                 f"Reference {from_rev} does not exist, considering entire history"
             )
```
Comment (danth) [inline semantic_release/vcs_helpers.py:50]: Generate the log message before changing the variable
```suggestion
            logger.debug(
                f"Reference {from_rev} does not exist, considering entire history"
            )
            from_rev = None
```

File: semantic_release/vcs_helpers.py [hunk 2]
```diff
@@ -37,18 +37,34 @@ def get_formatted_tag(version):
...
+    if to_rev:
+        to_rev = get_formatted_tag(to_rev)
+        try:
+            repo().commit(to_rev)
+        except BadName:
+            to_rev = None
+            logger.debug(
+                f"Reference {to_rev} does not exist, considering entire history until HEAD"
+            )
```
Comment (danth) [inline semantic_release/vcs_helpers.py:59]: Generate the log message before changing the variable
```suggestion
            logger.debug(
                f"Reference {to_rev} does not exist, considering entire history until HEAD"
            )
            to_rev = None
```

**Comment by danth** [comment]
The commitlint failure is because your commit message starts with a capital.

File: semantic_release/vcs_helpers.py (hunk 1 above)
Comment (betaboon) [reply semantic_release/vcs_helpers.py]: done

File: semantic_release/vcs_helpers.py (hunk 2 above)
Comment (betaboon) [reply semantic_release/vcs_helpers.py]: done

**Comment by betaboon** [comment]
> The commitlint failure is because your commit message starts with a capital.

That's funny. i was looking at the recent history on the repo to comply to the commit-messages.
The recent commits from @relekang seem to also break that commit-message rule :P

**Comment by danth** [comment]
We capitalise everything anyway when it goes into the changelog:

https://github.com/relekang/python-semantic-release/blob/5f28b835b0dbaaeaf2d80c6caaac3d66d4f7d0b0/semantic_release/history/logs.py#L126-L128

The linter is probably a bit too strict.

Anyway, these changes look good. Have you tested this in commit-mode too?

**Comment by betaboon** [comment]
> Anyway, these changes look good. Have you tested this in commit-mode too?

i tested this locally with `tag_only` + the current test-suite.
so no: didnt manually with commit-mode

**Comment by codejedi365** [comment]
I believe this PR might also solve #362 ?

**Comment by danth** [comment]
@betaboon Please can you rebase this to fix the conflicts. I will merge once that is done.

**Comment by betaboon** [comment]
I'm not close to a computer for the next couple of weeks.
So either you wait or do it yourself 😉

**Comment by ihowell-phaidra** [comment]
+1 to this branch. I am also being impacted by #362, which this solves.

**Comment by betaboon** [comment]
I just tried rebasing this branch, but there are some conflicts that I'm unsure about, that are the result of https://github.com/relekang/python-semantic-release/pull/488

**Pull request merged**